source/AI.cpp

```diff
@@ -3313,32 +3313,103 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 		// Board the nearest disabled ship, focusing on hostiles before allies. Holding
 		// `Shift` results in boarding only player-owned escorts in need of assistance.
 		shared_ptr<const Ship> target = ship.GetTargetShip();
-		if(!target || !CanBoard(ship, *target) || (shift && !target->IsYours()))
+		if(!target || (target && activeCommands.Has(Command::WAIT)) || !CanBoard(ship, *target) || (shift && !target->IsYours()))
 		{
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
-			bool foundAnything = false;
-			for(const shared_ptr<Ship> &other : ships)
-				if(CanBoard(ship, *other))
+			bool boardClosest = Preferences::Has("Board target");
+
+			auto strategy = [&]() -> function<double(const Ship &)>
+			{
+				Point current = ship.Position();
+				// If "board target" is set to false, value priority is used.
+				if(!boardClosest)
 				{
-					if(shift && !other->IsYours())
-						continue;
+					return [this, &ship](const Ship &other) -> double
+					{
+						// Use the exact cost if the ship was scanned, otherwise use an estimation.
+						return this->Has(ship, other.shared_from_this(), ShipEvent::SCAN_OUTFITS) ?
+							other.Cost() : (other.ChassisCost() * 2.);
```

Comment:
Why are we skipping over a ship right next to the flagship that is worth $10M for a ship halfway across the system that is worth $10.01M?

We absolutely want to consider distance and cost together, not in sequence.

Reply:
We do not skip any ships; we add all to the vector, to consider price later (also we need to add all of them so we can cycle them)

> We absolutely want to consider distance and cost together, not in sequence.

I agree, but that seems hard to do. The previous logic was giving weird results; and this is just intended to be a simple "take the best ship except if two are at the same distance" thing

Reply:
> edit:
> Now I get what you mean.
> That's how it works with distance preference, why would value be any different? This tie breaker is already more than distance does.

The chief difference is that distance-based boarding means your ship is not strafing the entire battlefield: it is always picking the closest thing.

> If you really want I can add said option, but `value` means "whatever happens give me the big ship" not "take the best option nearby"; that would be `best`.
> 2 coders playtested it; quuyykk and samenrocketman and they both thought it was a bug that a ship closer was taken instead of the big one.

What if we label the cost-weighted distance "mixed"? I don't think it is great piloting UX to have an option just for "cost," but exposing it too could certainly help shape users' expectations.

> So I think we need it as a third option, simply, that would combine both.
> I can easily do it in another PR, with the random option you wanted (I can't find any use for it but it won't be hard to code so I'll do it)

I'm not sure what you mean by "random option."

I had proposed one boarding strategy could be to just select the next ship in iteration order, but we don't need to do that.